components/automate-ui/src/app/modules/infra-proxy/tree-table/component/tree-table.component.ts

```diff
@@ -75,8 +75,10 @@ export class TreeTableComponent<T> implements OnInit, OnChanges {
     return new MatTableDataSource(this.treeTable.filter(x => x.isVisible));
   }
 
-  formatIndentation(node: TreeTableNode<T>, step: number = this.treeDepth): string {
-    return '&nbsp;'.repeat(node.depth * step);
+  formatIndentStyle(node: TreeTableNode<T>, step: number = this.steps) {
+    return {
+      'margin-left': (node.depth * step * step).toString() + 'px'
```

Comment:
Do we really want step-squared?

Reply:
While I like the approach of the solution, I'm noticing that this function runs in perpetuity.  It'd be great to either get the function out of the template, Or I found something like this here: https://stackoverflow.com/questions/57327412/angular-evaluating-ngstyle-expression-constantly-on-mouse-move

So that the function calls are limited -> not sure how this affects the rest of the page, but it solves the unnecessary calls unless @Inputs are changed

Reply:
without looking too deep in this, I think an alternate solution could be predefined CSS classes, applying the correct step class based on the step count.

Reply:
@vinay033 - I was thinking about this a little more.  is node.depth returning a number equivalent to how deep the item is nested?  If this is the case, we can likely dynamically add a class based on that number in the template, and then applied the required css off of that class.
something like
```
<div class="step-2">
```

and then in the scss
```
.somediv {
  &.step-1 {
    margin-left: 8px; // or whatever the px equivalent would be
  }

  &.step-2 {
    margin-left: 16px;
  }
}
```

Then we could remove that JS function all together

Reply:
@SEAjamieD , @msorens 
My point is that we are adding CSS via function because we don't know the depth of tree data and according to the above approach, we are restricted to apply CSS after tree depth reaching the steps.
currently, I have added changes as you suggested, for now, we are ok with this solution.